apps/files/src/components/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -98,8 +135,104 @@ export default {
       }
 
       return this.selectedRole
+    },
+
+    expirationSupported () {
+      return this.userExpirationDate && this.groupExpirationDate
+    },
+
+    defaultExpirationDateSet () {
+      if (this.editingUser) {
+        return this.userExpirationDate.enabled
+      }
+
+      if (this.editingGroup) {
+        return this.groupExpirationDate.enabled
+      }
+
+      return this.userExpirationDate.enabled || this.groupExpirationDate.enabled
+    },
+
+    userExpirationDate () {
+      return this.capabilities.files_sharing.user.expire_date
+    },
+
+    groupExpirationDate () {
+      return this.capabilities.files_sharing.group.expire_date
+    },
+
+    defaultExpirationDate () {
+      if (!this.defaultExpirationDateSet) {
+        return null
+      }
+
+      const userMaxExpirationDays = parseInt(this.userExpirationDate.days, 10)
+      const groupMaxExpirationDays = parseInt(this.groupExpirationDate.days, 10)
+
+      if (this.editingUser) {
+        return moment().add(userMaxExpirationDays, 'days').endOf('day').toISOString()
+      }
+
+      if (this.editingGroup) {
+        return moment().add(groupMaxExpirationDays, 'days').endOf('day').toISOString()
+      }
+
+      // Since we are not separating process for adding users and groups as collaborators
+      // we are using the one which is smaller as enforced date
+      let days = 0
+
+      if (userMaxExpirationDays && groupMaxExpirationDays) {
+        days = Math.min(userMaxExpirationDays, groupMaxExpirationDays)
+      } else {
+        days = userMaxExpirationDays || groupMaxExpirationDays
+      }
+
+      return moment().add(days, 'days').endOf('day').toISOString()
+    },
+
+    expirationDateEnforced () {
+      if (this.editingUser) {
+        return this.userExpirationDate.enforced
+      }
+
+      if (this.editingGroup) {
+        return this.groupExpirationDate.enforced
+      }
+
+      return this.userExpirationDate.enforced || this.groupExpirationDate.enforced
+    },
+
+    maxExpirationDate () {
+      if (!this.expirationDateEnforced) {
+        return null
+      }
+
+      return this.defaultExpirationDate
+    },
+
+    minExpirationDate () {
+      return moment().add(1, 'days').endOf('day').toISOString()
+    },
+
+    expirationDatePlaceholder () {
+      return this.$gettext('Expiration date')
+    },
+
+    expirationDateRemoveTooltip () {
+      return this.$gettext('Remove expiration date')
+    },
+
+    canResetExpirationDate () {
+      return !this.expirationDateEnforced && this.enteredExpirationDate
     }
   },
+
+  mounted () {
+    // FIXME: Datepicker is not displaying correct timezone so for now we add it manually
+    // this.enteredExpirationDate = this.expirationDate ? moment(this.expirationDate).toISOString(true) : this.defaultExpirationDate
```

Comment:
is it worthy to raise a tech debt ticket for this ? or too minor ?

Reply:
IMHO the whole date picker component should use some different library than it does now because it is modal. So I'm not sure if it's worth it to raise issue for this. Maybe a general issue for changing the library?